pkg/platform/openstack/loadbalancer.go

```diff
@@ -142,6 +142,26 @@ func ensureOpenStackLb(client *gophercloud.ServiceClient, name, vipAddress, vipS
 	if err != nil {
 		return "", errors.Wrap(err, "failed to extract LB list")
 	}
+
+	if len(lbs) == 0 && octaviaTagSupport {
+		// Attempt to retrieve API load balancer with description tagging
+		// to avoid another load balancer creation upon Octavia upgrade.
+		opts := loadbalancers.ListOpts{
+			Name:        name,
+			VipAddress:  vipAddress,
+			VipSubnetID: vipSubnetId,
+			Description: tag,
+		}
+		page, err = loadbalancers.List(client, opts).AllPages()
+		if err != nil {
+			return "", errors.Wrap(err, "failed to get LB list")
+		}
+		lbs, err = loadbalancers.ExtractLoadBalancers(page)
+		if err != nil {
+			return "", errors.Wrap(err, "failed to extract LB list")
+		}
+	}
```

Comment:
Shouldn't we tag it if we found it? That way on next loops we'll just make one call. Also it'll allow us to drop looking for LBs with tag in description after we'll drop support for Queens.

Reply:
Yup. Thanks for pointing this.

Reply:
Hm, on the other hand it's better for the user to take a look - otherwise it can end up with LB never updated. Okay, let's leave it as is.